### PR TITLE
90%: Add option to require profile upon login

### PR DESCRIPTION
**Pull request by pads**
This change allows for a consuming application to require that the user has a profile when they login.

An optional query string parameter is added to the generated login URL to inform Persona of this.

Contributes towards talis/platform#335

### CODE REVIEW

File: spec/unit/authentication/login_spec.rb [hunk 1]
```diff
@@ -27,6 +27,15 @@
         expect(@login.generate_url).to match expected_url
       end
 
+      it 'returns the correct login URL when a user profile is required' do
+        expected_url = %r{
+          http:\/\/persona\/auth\/providers\/trapdoor\/login
+          \?app=test-app&state=.*&redirectUri=http%3A%2F%2Fexample\.com
+          &require=profile
+        }x
+        expect(@login.generate_url(require_profile: true)).to match expected_url
```
Comment (omarqureshi) [inline spec/unit/authentication/login_spec.rb]: i prefer ```require: [:profile]``` or ```require: :profile```

Comment (omarqureshi) [inline spec/unit/authentication/login_spec.rb]: Are there other things that can be required?

Comment (pads) [inline spec/unit/authentication/login_spec.rb]: Not currently but I'm trying to future-proof. I'll pick option 1.

File: lib/talis/authentication/login.rb [hunk 2]
```diff
@@ -70,14 +70,19 @@ def initialize(app_id:, secret:, provider:, redirect_uri:)
...
         @state = Digest::MD5.hexdigest("#{@app}::#{@uuid.generate}")
-        params = URI.encode_www_form(
+        params = {
           app: @app,
           state: @state,
           redirectUri: @redirect_uri
-        )
+        }
+        params = params.merge(require: 'profile') if require_profile
+        params = URI.encode_www_form(params)
```
Comment (omarqureshi) [inline lib/talis/authentication/login.rb]: is params.to_query nicer? (do we include all of activesupport?)

File: spec/unit/authentication/login_spec.rb (hunk 1 above)
Comment (pads) [inline spec/unit/authentication/login_spec.rb]: Ugh, not gonna spend time on this - keeping it simple with option 2 for now. We're still pre-version 1 so we can refactor later as a breaking change.

**Comment by omarqureshi** [comment]
:+1:

**Pull request merged**